tests/test_part_image.py

```diff
@@ -238,9 +242,11 @@ def check_dir(dir):
 
                     mode = int(columns[2], 8)
                     # Recurse into directories.
-                    if mode & 040000 != 0 and columns[5] != "." and columns[5] != "..":
+                    if mode & 0o40000 != 0 and columns[5] != "." and columns[5] != "..":
                         check_dir(os.path.join(dir, columns[5]))
 
+                ls.wait()
```

Comment:
No, `wait` will not finish until `stdout` has been drained. I think this is correct, since we extract from `stdout` inside the loop now, instead of beforehand.